scripts/findUnused.ts

```diff
@@ -1,82 +1,33 @@
-import { relative } from "node:path"
-import { fromHere } from "@re-/node"
-import { BindingNamedNode, Node, Project, ts } from "ts-morph"
+import { join, relative } from "node:path"
+import { findPackageRoot, fromHere } from "@re-/node"
+import { BindingNamedNode, Project, SyntaxKind } from "ts-morph"
+import {
+    checkForIgnoreUnusedComment,
+    getPublicApiExports
+} from "./findUnusedHelpers.js"
 
 const project = new Project({
     tsConfigFilePath: fromHere("..", "tsconfig.references.json")
 })
 
-const unused: Record<string, string[]> = {}
-const snippetFileRegex = /snippets\/\w+/
-const exportAllRegex = /export \*/
+const unusedExports: Record<string, string[]> = {}
+const ignorePaths: RegExp[] = [new RegExp(join("docs", "snippets"))]
 const exportAllRenamedRegex = /\* as /
-const ignoreUnusedComment = "@ignore-unused"
-const apiExports: ApiExports[] = []
-const sourceFiles = project.getSourceFiles()
-type DeclarationInfo = {
-    filePath: string
-    sourceFileText: string
-    declarationNode: Node<ts.Node>
-    referencesLength: number
-    declarationName: string
-}
-type ApiExports = {
-    path: string
-    exportedDeclarations: string[]
-}
-const checkForIgnoreUnusedComment = ({
-    filePath,
-    sourceFileText,
-    declarationNode,
-    referencesLength,
-    declarationName
-}: DeclarationInfo) => {
-    for (const range of declarationNode.getLeadingCommentRanges()) {
-        if (
-            sourceFileText
-                .slice(range.getPos(), range.getEnd())
-                .includes(ignoreUnusedComment)
-        ) {
-            if (referencesLength > 1) {
-                console.log(
-                    `@ignore-unused wasn't needed for ${declarationName} - ${filePath}\n`
-                )
-            }
-            return true
-        }
-    }
-    return false
-}
 
-const indexFiles = sourceFiles.filter(
-    (file) => file.getBaseName() === "index.ts"
-)
-for (const file of indexFiles) {
-    const publicApiExports = file
-        .getExportDeclarations()
-        .filter((declaration) => !exportAllRegex.test(declaration.getText()))
-        .map((declaration) => {
-            return {
-                path: relative(".", declaration.getSourceFile().getFilePath()),
-                exportedDeclarations: declaration
-                    .getNamedExports()
-                    .map((namedExport) => namedExport.getName())
-            }
-        })
-    if (publicApiExports.length) {
-        apiExports.push(...publicApiExports)
-    }
-}
+const apiExports = getPublicApiExports(project)
```

Comment:
This looks good. I'd like to split up the rest of the script in a way that looks similar to this as well so that the top level just consists of a couple function calls.

On my branch, there is a new lint rule that limits functions to 32 lines. While I realize the fact that I have not merged that yet means it's kind of on me, it's probably a good heuristic overall to keep functions to around that length.